src/test/integration/tests/Rounding.t.sol

```diff
@@ -0,0 +1,277 @@
+    using StdStyle for *;
+
+    User attacker;
+    AVS badAVS;
+    IStrategy strategy;
+    IERC20Metadata token;
+    User goodStaker;
+    uint64 initTokenBalance;
+
+    OperatorSet mOpSet; // "manipOpSet" used for magnitude manipulation
+    OperatorSet rOpSet; // Redistributable opset used to exploit precision loss and trigger redistribution
+    
+    function _init() internal override {
+        _configAssetTypes(HOLDS_LST);
+
+        attacker = new User("Attacker"); // Attacker serves as both operator and staker
+        badAVS = new AVS("BadAVS"); // AVS is also attacker-controlled
+        strategy = lstStrats[0];
+        token = IERC20Metadata(address(strategy.underlyingToken()));
+        
+        // Prepares to add non-attacker stake into the protocol. Can be any amount > 0.
+        // Note that the honest stake does not need to be allocated anywhere, so long as it's in the same strategy.
+        goodStaker = new User("GoodStaker");
+        deal(address(token), address(goodStaker), uint256(1e18)); 
+        
+        // Register attacker as operator and create attacker-controlled AVS/OpSets
+        attacker.registerAsOperator(0);
+        rollForward({blocks: ALLOCATION_CONFIGURATION_DELAY + 1});
+        badAVS.updateAVSMetadataURI("https://example.com");
+        mOpSet = badAVS.createOperatorSet(strategy.toArray()); // setup low mag operator
+        rOpSet = badAVS.createRedistributingOperatorSet(strategy.toArray(), address(attacker)); // execute exploit
+
+        // Register for both opsets
+        attacker.registerForOperatorSet(mOpSet);
+        attacker.registerForOperatorSet(rOpSet);
+        
+        _print("setup");
+    }
+
+    // TODO: consider incremental manual fuzzing from 1 up to WAD - 1
+    function test_rounding_allMagsSlashed(uint16 slashes, uint64 initialWadToSlash, uint64 _initTokenBalance, uint24 r) public rand(r) {
+        vm.pauseGasMetering();
+        // Bound slashes to a reasonable range, with at least 1 slash.
+        // Note: Runs after 2500 hit OOG errors with default gas.
+        slashes = uint16(bound(slashes, 1, 5000));
+        
+        // We do two slashes, the sum of which slash 1 WAD (all operator magnitude) in total. 
+        // Each slash requires at least 1 mag. As such, we need to bound wadToSlash to 1 <= wadToSlash <= WAD - 1.
+        initialWadToSlash = uint64(bound(initialWadToSlash, 1, WAD - 1));
+        
+        // Bound initTokenBalance to a reasonable range to avoid overflow, with at least 1 token.
+        // Using ~18.45 quintillion tokens max (should be enough for any realistic test).
+        initTokenBalance = uint64(bound(_initTokenBalance, 1, type(uint64).max));
+        deal(address(token), address(attacker), initTokenBalance);
+        
+        _magnitudeManipulation(initialWadToSlash); // Manipulate operator magnitude for a given strategy.
+        _deposit(initialWadToSlash); // Setup operator with new opSet as well as honest stake in same strategy.
```

Comment:
Here's a point we may want to change - we're allocating all available magnitude before we use multiple slashes to "get funds out".

It kinda makes sense that this wouldn't result in any loss.

What if we don't allocate all available magnitude here, and instead allocate a portion of it (parameterized). We then use multiple slashes to get a portion of the funds out via redistribution, and then attempt to withdraw remaining funds via queue+completeWithdrawals?